### ISSUE TITLE
Gaiacli incorrectly parses configured URL

### ISSUE THREAD

**Issue by feri42** [cosmos/gaia#322]
## Summary of Bug

Gaiacli incorrectly parses the URL from the configuration file. Example the URL:
```
gaiacli config node https://a.b.example.com:443/gaiad
# The URL is correctly saved in ~/.gaiacli/config/config.toml
gaiacli status
ERROR: Status: parse "http://a.b.example.com:443.gaiad": invalid port ":443.gaiad" after host
```

## Version

2.0.4
2.0.6

## Steps to Reproduce

```
gaiacli config node https://a.b.example.com:443/gaiad
gaiacli status
ERROR: Status: parse "http://a.b.example.com:443.gaiad": invalid port ":443.gaiad" after host
```
Not only it is confused by the '/' after the host part, it ignores the HTTPS protocol part.
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by tac0turtle** [comment]
@feri42 if you update to 2.0.8 are you still running into this issue?

**Comment by gin** [comment]
✅ The HTTPS protocol part is fixed in v2.0.5 and after (It works in v2.0.6 too).
❌ However, the path after the port number is still incorrectly parsed in v2.0.8, and latest v2.0.10.

Workaround for users (@feri42) could be to set a CNAME in the domain's resource records. e.g. `https://gaiad.example.com:443`

**Comment by tac0turtle** [comment]
the parsing of urls don't work with `:443.gaiad`. You would need to make an issue on the sdk or tendermint to support this.
